### ISSUE TITLE
MapDataset.to_spectrum_dataset produces aeff with incorrect unit

### ISSUE THREAD

**Issue by lmohrmann** [gammapy/gammapy#2693]
**Gammapy version**
0.15

**Bug description**
When producing a `SpectrumDataset` with `MapDataset.to_spectrum_dataset()`, the `aeff` attribute of the `SpectrumDataset` has unit 1/s.

**Expected behavior**
The dimension of the `aeff` attribute should (obviously) be area, i.e. unit should be (e.g.) m^2.

**To Reproduce**
Run the analysis_3d.ipynb tutorial notebook, and at the end do:
```
spec_ds = analysis_joint.datasets[0].to_spectrum_dataset(region)
spec_ds.aeff.data.data
```

**Related Question**
I noticed that the created `SpectrumDataset` has no model attached. Wouldn't it make sense to transfer the `spectral_model` of the `MapDataset` to the `SpectrumDataset`?

**Comment by registerrier** [comment]
Thanks @lmohrmann . Indeed, the unit is not correctly passed and this was not properly tested. 
Adding a correction now.

Regarding the `model`, we use `SkyModels` also for `SpectrumDatasetOnOff`. So we could simply tell the function which model component to keep, as we do for `FluxPointsEstimator`.
Opinion @adonath ?

**Comment by adonath** [comment]
@registerrier @lmohrmann  I'm not sure such a minimal option to keep the model is really helpful for now, just because it can already be achieved with one line like `spectrum_dataset.models = dataset.models["my-source"]`

If we add an option to keep (and reduce) the 3D model, I would argue it should be more helpful, e.g. compute the contributions of the remaining model components in the analysis region and keep as fixed background contributions in the 1D model. But there it quickly get's complicated...

I think for now I would add a comment in the docstring mentioning, that the model is dropped and   show how to select the model component after. Later we improve to something similar I mentioned above (i.e. compute "contamination" contributions and keep this in the 1D model as background components).

**Comment by lmohrmann** [comment]
@adonath That's also fine with me, but it would be good to document this. We used `plot_fit` on the created `SpectrumDataset` and were wondering why `mu_src` was well below the Excess - the reason was precisely that the model was not propagated, and only the background was evaluated for `mu_src`.

**Comment by adonath** [comment]
@lmohrmann Thats's a reasonable use-case we (honestly) didn't thought of, when we introduced the `.to_spectrum_dataset()` method. Initially it was only meant to produce a dataset for spectral fits of extended sources, but indeed you can use it to inspect the spectral parts of 3D models in certain regions as well. So we'll add a comment in the docstring for now.

**Comment by registerrier** [comment]
This is solved by PR #2694 . Closing issue for now.

Exporting the model to get correct `mu_src` requires an evaluation of the models on the `MapDataset.geom` and the creation of new spectral models integrating the the spatial part properly. This might be done. But how to do it with complex multicomponent models (or `SkyDiffuseCube` models) should be clarified. 

Note that this can be way to deal with PSF containment for non circular regions and leaking fraction for extended sources.  Moving this to a new independent issue.
